Vocabulary update fails

Tried to add some tokens to vocabulary (update_vocab: true, reset_optim: all), eole acknowledges added tokens:
[2025-01-14 16:51:05,854 INFO] src: 5120 new tokens

Then:
-- Tracebacks above this line can probably
                 be ignored --

Traceback (most recent call last):
  File "/nfs/onmt/eole-003/eole/eole/utils/distributed_workers.py", line 48, in spawned_train
    process_fn(config, device_id=device_id)
  File "/nfs/onmt/eole-003/eole/eole/train_single.py", line 181, in main
    model, _, _ = get_model_class(config.model).from_config(
  File "/nfs/onmt/eole-003/eole/eole/models/model.py", line 455, in from_config
    model.training_logic(running_config, vocabs, checkpoint, device_id)
  File "/nfs/onmt/eole-003/eole/eole/models/model.py", line 287, in training_logic
    self.load_checkpoint(running_config, vocabs, checkpoint, device, offset)
  File "/nfs/onmt/eole-003/eole/eole/models/model.py", line 261, in load_checkpoint
    self.load_safe_state_dict(
  File "/nfs/onmt/eole-003/eole/eole/models/model.py", line 670, in load_safe_state_dict
    self._load_param(
  File "/nfs/onmt/eole-003/eole/eole/models/model.py", line 531, in _load_param
    param.data.size()
AssertionError: An error in model's partition and checkpoint's slice was detected, [src_emb.embeddings, Embedding(136192, 2048, padding_idx=1), weight, torch.Size([136192, 2048]), torch.Size([131072, 2048])]

(original vocab had 131072 tokens)

--- a/eole/models/model.py
+++ b/eole/models/model.py
@@ -628,7 +628,7 @@ class BaseModel(nn.Module):
                         ckpt_t = updated_params[name + "." + param_name]
                         self._load_param(name, module, param_name, param, buf_list, ckpt_t, offset)
                         keyfound[name + "." + param_name] = True
-                    if name + "." + param_name in keys_shard.keys():
+                    elif name + "." + param_name in keys_shard.keys():
 
                         ckpt_t = f[keys_shard[name + "." + param_name]].get_tensor(name + "." + param_name)
                         self._load_param(name, module, param_name, param, buf_list, ckpt_t, offset)

